hail/src/main/scala/is/hail/expr/types/MatrixType.scala

```diff
@@ -18,6 +19,12 @@ class MatrixTypeSerializer extends CustomSerializer[MatrixType](format => (
 object MatrixType {
   val entriesIdentifier = "the entries! [877f12a8827e18f61222c6c8c5fb04a8]"
 
+  def getRowType(rvRowType: PStruct): PStruct = rvRowType.selectFields(Set(entriesIdentifier), keep = false)
+  def getEntryArrayType(rvRowType: PStruct): PArray = rvRowType.field(entriesIdentifier).typ.asInstanceOf[PArray]
+  def getEntriesType(rvRowType: PStruct): PStruct = rvRowType.selectFields(Set(entriesIdentifier), keep = true)
```

Comment:
Can we get a better name for this?  It reads like it should be the same as `gtEntriesArrayType` (except has return type `Type`)

Reply:
is `getSplitEntriesType` ok?

Reply:
I probably would have included "Row" in there somewhere, but this is more than good enough for me.  Thanks!